samples/hello-cube.html

```diff
@@ -0,0 +1,336 @@
+    // const depthStateDescriptor = {
+    //     boolean depthWriteEnabled: true,
+    //     *GPUCompareFunction depthCompare: "less",
+
+    //     *GPUStencilStateFaceDescriptor front;
+    //     *GPUStencilStateFaceDescriptor back;
+
+    //     u32 stencilReadMask;
+    //     u32 stencilWriteMask;
+    // };
+
+    // Pipeline
+    const pipelineLayoutDescriptor = { bindGroupLayouts: [uniformsBindGroupLayout] };
+    const pipelineLayout = device.createPipelineLayout(pipelineLayoutDescriptor);
+    const vertexStageDescriptor = {
+        module: shaderModule,
+        entryPoint: "vertex_main"
+    };
+    const fragmentStageDescriptor = {
+        module: shaderModule,
+        entryPoint: "fragment_main"
+    };
+    const pipelineDescriptor = {
+        layout: pipelineLayout,
+
+        vertexStage: vertexStageDescriptor,
+        fragmentStage: fragmentStageDescriptor,
+
+        primitiveTopology: "triangleList",
+        // *GPURasterizationStateDescriptor rasterizationState;
+        // sequence<*GPUBlendStateDescriptor> blendStates;
+        // *GPUDepthStencilStateDescriptor depthStencilState: depthStateDescriptor
+        inputState: inputStateDescriptor
+        // *GPUAttachmentsStateDescriptor attachmentsState;
+ 
+        // u32 sampleCount; // Multisampling
+    };
+    pipeline = device.createRenderPipeline(pipelineDescriptor);
+
+    let colorAttachment = {
+        // attachment is acquired in render loop.
+
+        // GPUTextureView? resolveTarget;
+
+        // *GPULoadOp loadOp;
+        // *GPUStoreOp storeOp;
+        clearColor: { r: 0.5, g: 1.0, b: 1.0, a: 1.0 } // GPUColor
+    };
+
+    // Depth stencil texture
+
+    // *GPUExtent3D
+    // const depthSize = {
+    //     u32 width;
+    //     u32 height;
+    //     u32 depth;
+    // };
+
+    // *GPUTextureDescriptor
+    // const depthTextureDescriptor = {
+    //     *GPUExtent3D size: depthSize,
+    //     u32 arraySize;
+    //     u32 levelCount;
+    //     u32 sampleCount: 1,
+    //     *GPUTextureDimension dimension: "2d",
+    //     *GPUTextureFormat format: "D32FloatS8Uint",
+    //     *GPUTextureUsageFlags usage: GPUTextureUsage.OUTPUT_ATTACHMENT
+    // };
+
+    // GPUTexture + *GPUDevice::createTexture()
+    // const depthTexture = device.createTexture(depthTextureDescriptor);
+
+    // *GPURenderPassDepthStencilAttachmentDescriptor
+    // const depthAttachment = {
+    //     GPUTextureView attachment: depthTexture
+
+    //     *GPULoadOp depthLoadOp;
+    //     *GPUStoreOp depthStoreOp;
+    //     float clearDepth: 1.0;
+
+    //     *GPULoadOp stencilLoadOp;
+    //     *GPUStoreOp stencilStoreOp;
+    //     u32 clearStencil;
+    // };
+
+    renderPassDescriptor = { 
+        colorAttachments: [colorAttachment] 
+        // *depthStencilAttachment: depthAttachment
+    };
+
+    queue = device.getQueue();
+
+    render();
+}
+
+function render() {
+    populateTransformArray(mappedTransformArray);
```

Comment:
Don't you need a fence here or multiple buffers before you can modify the mapped transform array, given that it might still be in-flight from the previous frame?

Reply:
[1] explicitly calls out that this will need to be updated once #147 is landed.